### PR TITLE
Benchmarkapp: Added processing of inputs/outputs by index

### PR DESCRIPTION
**Pull request by fzhar**
inputs/outputs precision processing algorithm was refactored to handle possible no-tensor-name issues

### CODE REVIEW

File: samples/cpp/benchmark_app/main.cpp [hunk 1]
```diff
@@ -472,39 +472,58 @@ int main(int argc, char* argv[]) {
...
+            for (int i = 0; i < inputs.size(); i++) {
+                const auto& item = inputs[i];
+                auto iop_precision = ov::element::undefined;
+                auto type_to_set = ov::element::undefined;
+                std::string name;
+                try {
+                    // Some tensors might have no names, get_friendly_name will throw exception in that case.
+                    // -iop option will not work for those tensors.
+                    name = item.get_any_name();
+                    iop_precision = getPrecision2(user_precisions_map.at(item.get_node()->get_friendly_name()));
```
Comment (ilyachur) [inline samples/cpp/benchmark_app/main.cpp:485]: ?! Why do you need this code? OV 2.0 doesn't work with operation name. Please use name variable from the line 484.

File: samples/cpp/benchmark_app/main.cpp [hunk 2]
```diff
@@ -472,39 +472,58 @@ int main(int argc, char* argv[]) {
...
-                    const auto precision = getPrecision2(user_precisions_map.at(name));
-                    preproc.output(name).tensor().set_element_type(precision);
+            const auto& outs = model->outputs();
+            for (int i = 0; i < outs.size(); i++) {
+                const auto& item = outs[i];
+                auto iop_precision = ov::element::undefined;
+                try {
+                    // Some tensors might have no names, get_friendly_name will throw exception in that case. 
+                    // -iop option will not work for those tensors. 
+                    iop_precision = getPrecision2(user_precisions_map.at(item.get_node()->get_friendly_name()));
```
Comment (ilyachur) [inline samples/cpp/benchmark_app/main.cpp:519]: Please use tensor names

File: samples/cpp/benchmark_app/main.cpp (hunk 1 above)
Comment (fzhar) [reply samples/cpp/benchmark_app/main.cpp]: fixed

File: samples/cpp/benchmark_app/main.cpp (hunk 2 above)
Comment (fzhar) [reply samples/cpp/benchmark_app/main.cpp]: fixed

**Review by vladimir-dudnik — APPROVED** [review]

**Pull request merged**